plugins/modules/foreman_hostgroup.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2019 Manisha Singhal (ATIX AG)
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: foreman_hostgroup
+short_description: Manage Foreman Hostgroups using Foreman API
+description:
+  - Create, Update and Delete Foreman Hostgroups using Foreman API
+author:
+  - "Manisha Singhal (@Manisha15) ATIX AG"
+requirements:
+  - "apypie"
+options:
+  name:
+    description: Name of Hostgroup
+    required: true
+  description:
+    description: description of hostgroup
+    required: false
+  locations:
+    description: List of locations the hostgroup should be assigned to
+    required: false
+    default: None
+    type: list
+  organizations:
+    description: List of organizations the hostgroup should be assigned to
+    required: false
+    default: None
+    type: list
+  architecture:
+    description: Architecture name
+    required: False
+    default: None
+  domain:
+    description: Domain name
+    required: False
+    default: None
+  environment:
+    description: Environment name
```

Comment:
```suggestion
    description: Puppet environment name
```

(I know, the param is called just `environment`, but it can be confusing when  you also have Katello installed which has Lifecycle Environments)